### PR TITLE
Accelerating unbalanced OT

### LINKED ISSUE #196: Implement translation invariant Sinkhorn update
[paper](https://proceedings.mlr.press/v151/sejourne22a.html) (algorithm 2), related to #194

### PR DESCRIPTION
**Pull request by michalk8**
TODOs:
- [x] ~~figure out why 0s in marginals make it not converge (convergence criterion)?~~ was mostly an issue with the geometry (Grid) when testing this, seems to be ok, will test
- [x] handle case when `tau_a=1.0` or `tau_b=1.0`: UPDATE: no acceleration done in this case
- [x] ~~raise error when not using LSE mode (or implement in kernel mode)~~ disabled in kernel mode + updated docstrings
- [x] update docs + rename `normalize_potentials`
- [x] tests

closes #196

### CODE REVIEW

File: src/ott/solvers/linear/sinkhorn.py [hunk 1]
```diff
@@ -61,6 +72,28 @@ def ent_reg_cost(
   ) -> float:
     return ent_reg_cost(self.fu, self.gv, ot_prob, lse_mode)
 
+  def shift(
```
Comment (marcocuturi) [inline src/ott/solvers/linear/sinkhorn.py:75]: I like `shift` because this is what we do, but WDYT of `recenter`?

File: src/ott/solvers/linear/sinkhorn.py [hunk 2]
```diff
@@ -103,19 +136,19 @@ def solution_error(
   # and h is either f or g. Note this is equal to z if rho_z → inf, which
   # is the case when tau_z → 1.0
   if lse_mode:
```
Comment (marcocuturi) [inline src/ott/solvers/linear/sinkhorn.py:138]: it seems I forgot to add `only_check_b` in the list of args above when I introduced it in a previous PR. Could you correct this?

File: src/ott/solvers/linear/sinkhorn.py [hunk 3]
```diff
@@ -400,6 +427,7 @@ def __init__(
       momentum: Optional[acceleration.Momentum] = None,
       anderson: Optional[acceleration.AndersonAcceleration] = None,
       parallel_dual_updates: bool = False,
+      normalize_potentials: bool = False,
```
Comment (marcocuturi) [inline src/ott/solvers/linear/sinkhorn.py:430]: docstring missing

File: src/ott/solvers/linear/sinkhorn.py [hunk 4]
```diff
@@ -626,14 +691,12 @@ def output_from_state(
...
+
     f = state.fu if self.lse_mode else geom.potential_from_scaling(state.fu)
     g = state.gv if self.lse_mode else geom.potential_from_scaling(state.gv)
-    if ot_prob.is_balanced:
-      # center the potentials for numerical stability if the problem is balanced
-      is_finite = jnp.isfinite(f)
-      center = jnp.sum(jnp.where(is_finite, f, 0.)) / jnp.sum(is_finite)
-      f -= center
-      g += center
+    if self.normalize_potentials:
```
Comment (marcocuturi) [inline src/ott/solvers/linear/sinkhorn.py:697]: maybe `recenter_potentials` is more self-explanatory?

**Review by marcocuturi — COMMENTED** [review]
thanks Michal, this looks cool!

File: src/ott/solvers/linear/sinkhorn.py (hunk 3 above)
Comment (michalk8) [reply src/ott/solvers/linear/sinkhorn.py]: Done.

**Review by marcocuturi — APPROVED** [review]
Thanks! this really great and useful!!
